source/GameInterface/Services/MapEvents/Patches/EnterSettlementPatch.cs

```diff
@@ -0,0 +1,23 @@
+﻿using Common.Messaging;
+using Common.Util;
+using Common;
+using HarmonyLib;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using TaleWorlds.CampaignSystem.Encounters;
+using TaleWorlds.CampaignSystem.Party;
+
+namespace GameInterface.Services.MapEvents.Patches
+{
+    [HarmonyPatch(typeof(PlayerEncounter), "EnterSettlement")]
+    public class EnterSettlementPatch
+    {
+        static bool Prefix()
+        {
+            MessageBroker.Instance.Publish(MobileParty.MainParty, new SettlementEntered(MobileParty.MainParty.TargetSettlement.StringId, MobileParty.MainParty.StringId));
```

Comment:
Create message object on different line
```suggestion
            var message = new SettlementEntered(MobileParty.MainParty.TargetSettlement.StringId, MobileParty.MainParty.StringId);
            MessageBroker.Instance.Publish(MobileParty.MainParty, message);
```